src/pages/SignUp/SignUp.js

```diff
@@ -1,8 +1,174 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faChevronDown } from '@fortawesome/free-solid-svg-icons'
+import { TERMS_BOX } from './termsBox'
+import './SignUp.scss'
 
 const SignUp = () => {
-  return <h1>hello</h1>
+  const [email, setEmail] = useState('')
+  const [emailError, setEmailError] = useState(false)
+  const [name, setName] = useState('')
+  const [nameError, setNameError] = useState(false)
+  const [termsOn, setTermsOn] = useState([false, false, false])
+  const [checkItems, setCheckItems] = useState([])
+
+  const handleSingleCheck = (checked, id) => {
+    if (checked) {
+      // 단일 선택 시 체크된 아이템을 배열에 추가
+      setCheckItems(prev => [...prev, id])
+    } else {
+      // 단일 선택 해제 시 체크된 아이템을 제외한 배열 (필터)
+      setCheckItems(checkItems.filter(el => el !== id))
+    }
+  }
+
+  const handleAllCheck = checked => {
+    if (checked) {
+      // 전체 선택 클릭 시 데이터의 모든 아이템(id)를 담은 배열로 checkItems 상태 업데이트
+      const idArray = []
+      TERMS_BOX.forEach((el, index) => idArray.push(index))
+      setCheckItems(idArray)
+    } else {
+      // 전체 선택 해제 시 checkItems 를 빈 배열로 상태 업데이트
+      setCheckItems([])
+    }
+  }
+
+  return (
+    <div className="signUp">
+      <h1 className="signUpTitle">회원가입</h1>
+      <div className="infTitle">회원정보</div>
+      <ul className="signUpInputContainer">
+        <li>
+          <input
+            autoFocus={true}
+            className={emailError ? 'inputError' : 'input'}
+            type="text"
+            placeholder="이메일"
+            onChange={e => {
+              setEmail(e.target.value)
+              {
+                e.target.value === ''
+                  ? setEmailError(true)
+                  : setEmailError(false)
+              }
+            }}
+          ></input>
+        </li>
+        {emailError ? (
+          <p className="errorMsg">이메일은 필수 입력값입니다.</p>
+        ) : null}
```

Comment:
```suggestion
        {emailError && (
          <p className="errorMsg">이메일은 필수 입력값입니다.</p>
        )}
```
조건이 false일 때, UI가 보이지 않아도 된다면 && 연산자를 사용해주세요!